src/App/Fossa/Main.hs

```diff
@@ -150,14 +153,30 @@ mergeFileCmdConfig cmd file =
     , optProjectName = optProjectName cmd <|> (configProject file >>= configProjID)
     , optProjectRevision = optProjectRevision cmd <|> (configRevision file >>= configCommit)
     , optAPIKey = optAPIKey cmd <|> configApiKey file
+    , optConfig = optConfig cmd
     , optCommand = optCommand cmd
     }
 
 appMain :: IO ()
 appMain = do
   initCapabilities
   cmdConfig <- customExecParser mainPrefs (info (opts <**> helper) (fullDesc <> header "fossa-cli - Flexible, performant dependency analysis"))
-  fileConfig <- readConfigFileIO
+
+  fileConfigPath <-
+    case (optConfig cmdConfig) of
+      Just userProvidedFile -> do sendIO $ validateFile userProvidedFile
+      Nothing ->
+        case optCommand cmdConfig of
+          AnalyzeCommand AnalyzeOptions{analyzeBaseDir} -> do
+            baseDir <- sendIO $ validateDir analyzeBaseDir
+            pure (unBaseDir baseDir </> defaultConfigFile)
+          TestCommand TestOptions{testBaseDir} -> do
+            baseDir <- sendIO $ validateDir testBaseDir
+            pure (unBaseDir baseDir </> defaultConfigFile)
+          _ -> do
+            cwd <- getCurrentDir
+            pure (cwd </> defaultConfigFile)
+  fileConfig <- readConfigFileIO fileConfigPath
```

Comment:
Switching the behavior of where the default config file is located is a breaking change.  It's possible that this breaks existing workflows that rely on using the current working dir.

I'm not sure we actually want to make this change at the moment.  Unless we can get team agreement on breaking this, we should probably leave it as-is.  Also, it's not necessary for this PR, so if you decide you want to not wait for that discussion to get this merged, you could remove it and postpone it.

Reply:
I reverted to using cwd like before for config file in 599d282d9bf96e5ca52ffd787a76a404e4680b4a. 

I rather get this merged sooner, but I do think we should eventually look for config file at target directory of command, as opposed to cwd.